arbnode/batch_poster.go

```diff
@@ -427,6 +463,39 @@ func AccessList(opts *AccessListOpts) types.AccessList {
 	return l
 }
 
+// Adds a block merkle proof to an Espresso justification, providing a proof that a set of transactions
+// hashes to some light client state root
+func (b *BatchPoster) addEspressoBlockMerkleProof(
+	msg *arbostypes.MessageWithMetadata,
+) (*arbostypes.MessageWithMetadata, error) {
+	if msg.Message.Header.Kind == arbostypes.L1MessageType_L2Message &&
+		msg.Message.L2msg[0] == arbos.L2MessageKind_EspressoTx {
```

Comment:
This check happens in multiple places now, we could consider extracting a helper function for it.